Modules/_zstd/compressor.c

```diff
@@ -289,14 +282,14 @@ _zstd_load_c_dict(ZstdCompressor *self, PyObject *dict)
     /* Check (ZstdDict, type) */
     if (PyTuple_CheckExact(dict) && PyTuple_GET_SIZE(dict) == 2) {
         /* Check ZstdDict */
-        ret = PyObject_IsInstance(PyTuple_GET_ITEM(dict, 0),
-                                  (PyObject*)mod_state->ZstdDict_type);
-        if (ret < 0) {
-            return -1;
-        }
-        else if (ret > 0) {
-            /* type == -1 may indicate an error. */
+        if (PyObject_TypeCheck(PyTuple_GET_ITEM(dict, 0),
+                               mod_state->ZstdDict_type) &&
+            PyLong_Check(PyTuple_GET_ITEM(dict, 1)))
```

Comment:
Binary operators first, per PEP 7?
```suggestion
                               mod_state->ZstdDict_type)
            && PyLong_Check(PyTuple_GET_ITEM(dict, 1)))
```